File: leetcode-submissions/209_Minimum_Size_Subarray_Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        //now we are using sliding window technique in which we are using 
        int j=0;
        int ans=INT_MAX;
        int sum=0;
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            while(j<=i and sum>=target){
                ans=min(ans,i-j+1);
                sum-=nums[j];
                j++;
            }
        }
        return ans==INT_MAX?0:ans;

        
    }
};

File: leetcode-submissions/209_Minimum_Size_Subarray_Sum.ts
function minSubArrayLen(target: number, nums: number[]): number {
    let left =0;
    let right :number =0;
    let n :number =nums.length ;
    let minLength :number =Number.MAX_SAFE_INTEGER;


        let sum :number=0;
    while(right<n){
        sum+=nums[right];
        //adding value to the answer until the sum is exeeds 

        while ( sum>=target ){
            // need to add it to the answer and calculate the minans 
            sum-=nums[left];
            minLength=Math.min(minLength,right-left+1);
            left++;
        }
right ++;
    }
    return minLength===Number.MAX_SAFE_INTEGER?0:maxLength;
};